app/gui/src/model/module/synchronized.rs

```diff
@@ -502,52 +493,63 @@ impl Module {
     /// might not be the minimal diff, but will contain all changes.
     #[profile(Debug)]
     fn partial_invalidation(
         &self,
-        ls_content: &ParsedContentSummary,
+        ls_content: ParsedContentSummary,
         new_file: SourceFile,
-    ) -> impl Future<Output = FallibleResult<ParsedContentSummary>> + 'static {
+    ) -> impl Future<Output = ()> + 'static {
         debug!("Handling partial invalidation: {ls_content:?}.");
         let edits = vec![
             //id_map and metadata go first, because code change may alter their position.
-            Self::edit_for_idmap(ls_content, &new_file),
-            Self::edit_for_metadata(ls_content, &new_file),
-            Self::edit_for_code(ls_content, &new_file),
+            Self::edit_for_idmap(&ls_content, &new_file),
+            Self::edit_for_metadata(&ls_content, &new_file),
+            Self::edit_for_code(&ls_content, &new_file),
         ]
         .into_iter()
         .flatten()
         .collect_vec();
-        self.notify_language_server(&ls_content.summary, &new_file, edits, true)
+        self.notify_language_server(ls_content.summary.digest, &new_file, edits, true)
     }
 
     /// This is a helper function with all common logic regarding sending the update to
-    /// Language Server. Returns the new summary of Language Server state.
+    /// Language Server.
+    ///
+    /// The current value of `Self::ls_content` field s not used in this function, but replaced
+    /// with the new one basing on `new_file` argument.
     #[profile(Debug)]
     fn notify_language_server(
         &self,
-        ls_content: &ContentSummary,
+        ls_content_digest: Sha3_224,
         new_file: &SourceFile,
         edits: Vec<TextEdit>,
         execute: bool,
-    ) -> impl Future<Output = FallibleResult<ParsedContentSummary>> + 'static {
+    ) -> impl Future<Output = ()> + 'static {
         let summary = ParsedContentSummary::from_source(new_file);
         let edit = FileEdit {
             edits,
             path: self.path().file_path().clone(),
-            old_version: ls_content.digest.clone(),
+            old_version: ls_content_digest,
             new_version: Sha3_224::new(new_file.content.as_bytes()),
         };
-        debug!("Notifying LS with edit: {edit:#?}.");
         let ls_future_reply = self.language_server.client.apply_text_file_edit(&edit, &execute);
+        let ls_content = self.ls_content.clone_ref();
         async move {
-            ls_future_reply.await?;
-            Ok(summary)
+            debug!("Notifying LS with edit: {edit:#?}.");
+            match ls_future_reply.await {
+                Ok(()) => {
+                    debug!("Updating the LS content digest to: {:?}", summary);
+                    ls_content.replace(LanguageServerContent::Synchronized(summary));
+                }
+                Err(err) => {
+                    error!("Error during sending text change to Language Server: {err}");
```

Comment:
Minor inconsistency - some errors use `LS`, some use `Language Server`. I think that abbreviations are always worse than full names, so I'd be thankful for moving them to full names instead.